raiden/tests/integration/test_send_queued_messages.py

```diff
@@ -165,25 +166,55 @@ def test_payment_statuses_are_restored(
 
     app0.event_handler = HoldRaidenEvent()
     app0.event_handler.hold(SendSecretReveal, {})
+    app2.event_handler = HoldRaidenEvent()
+    app2.event_handler.hold(SendSecretRequest, {})
+
+    # make one transfer from app0 to app2 so that the locked amount
+    # in the app0-app1 channel also includes collateral from other channels
+    identifier = 42
+    payment_status = app0.raiden.mediated_transfer_async(
+        token_network_identifier=token_network_identifier,
+        amount=2,
+        target=app2.raiden.address,
+        identifier=identifier,
+    )
+    assert payment_status.payment_identifier == identifier
 
     # make a few transfers from app0 to app1
     amount = 1
     spent_amount = 7
     identifier = 1
 
-    for _ in range(spent_amount):
+    for identifier in range(spent_amount):
+        identifier = identifier + 1
         payment_status = app0.raiden.mediated_transfer_async(
             token_network_identifier=token_network_identifier,
             amount=amount,
             target=app1.raiden.address,
             identifier=identifier,
         )
         assert payment_status.payment_identifier == identifier
-        identifier += 1
 
     raiden_event_handler = RaidenEventHandler()
     message_handler = MessageHandler()
-
+    # Bring down both app0 and app1 so that we can check app0's payment statuses
+    # during restart being sure that the payments won't complete (and pop the mapping)
+    # before we are done checking
+    app1_restart = App(
```

Comment:
I believe just calling `stop` here and then `start` bellow is sufficient. IOW, I don't think you need to create a new instance of `app1`

Reply:
Sure I can do that. Problem is I had no way to test it. Even without doing the stopping of app1 it works in my machine, but can fail due to the described race in a very fast machine like CircleCI.

Reply:
I don't follow, `stop` should be synchronous.

Reply:
Never mind